### ISSUE TITLE
Error: nested form - Expected type Date for field “Date”, received “2019-02-21T13:00:00.000Z” instead

### ISSUE THREAD

**Issue by bseng** [VulcanJS/Vulcan#2211]
vulcan version : 1.12.12

Steps to replicate:

- Create a schema with a nested schema using Simple Schema (How it was done in example-form).
- Set one of the field in the nested schema to be of type date.
- Create a smartform for the schema.
- Set any date in the field in the nested schema.

Expected
- Form saved successfully.

Actual
- Error: Expected type Date for field “Date”, received “2019-02-08T13:00:00.000Z” instead.

**Comment by Apollinaire** [comment]
I think I've had this problem before and just found a workaround to not use dates in nested schemas. Not optimal, probably something that we should fix & test with Apollo 2.
IIRC it was related to JSON not handling dates.

**Comment by eric-burel** [comment]
@SachaG I tried to fix this but could not understand where date are handled? There is a "convertDate" util but it seems unused (and indeed does not support nesting).

Solution seems to convert all dates back to JS object before validating. If we need to iterate on the schema we could reuse the vulcan forms `schema_utils` and create a generic helpers that takes nesting into account.

**Comment by eric-burel** [comment]
Closing as this issue will be resolved in the incoming `1.14` version. If you don't use custom schemas, the issue will simply disappear (tag me if it's not the case and I'll reopen).

If you use custom schemas, you now need to add your Date field in your custom schema like this:
```
{
     event {
          startDate
     }
}
```

To sum it up GraphQL handles the scalar fields (think native JS objects like `Date`) automatically. However in previous of Vulcan, we did not generate the full graphQL schema for nested field, so the conversion did not happen. And thus SimpleSchema did not receive a JS `Date` object as expected but a string and used to throw this error (basically it was a far more complicated issue than what it seemed to be ^^).
Now the schema is correctly generated.
